android/libraries/rib-android/src/main/kotlin/com/uber/rib/core/XRay.kt

```diff
@@ -23,70 +23,74 @@ import android.graphics.drawable.BitmapDrawable
 import android.graphics.drawable.Drawable
 import android.view.Gravity
 import android.view.View
+import androidx.annotation.VisibleForTesting
 
 /** Utility class that shows riblets name in its background. */
 class XRay private constructor() {
-  private var isEnabled = false
-  private var textPaint: Paint? = null
-  private fun writeOnBitmap(bitmap: Bitmap, text: String) {
-    val canvas = Canvas(bitmap)
-    val textPaint = getTextPaint()
-    val xStartPoint = (bitmap.width - textPaint.measureText(text)) / 2f
-    val yStartPoint = bitmap.height / 2f
-    canvas.drawText(text, xStartPoint, yStartPoint, textPaint)
+  private var config = XRayConfig()
+  private val textPaint: Paint by lazy {
+    Paint().apply {
+      textSize = TEXT_SIZE.toFloat()
+      color = TEXT_COLOR
+    }
   }
 
-  private fun getTextPaint(): Paint {
-    if (textPaint == null) {
-      textPaint =
-        Paint().apply {
-          textSize = TEXT_SIZE.toFloat()
-          color = TEXT_COLOR
-        }
+  private fun writeOnBitmap(bitmap: Bitmap, text: String) {
+    Canvas(bitmap).run {
+      val xStartPoint = (bitmap.width - textPaint.measureText(text)) / 2f
+      val yStartPoint = bitmap.height / 2f
+      drawText(text, xStartPoint, yStartPoint, textPaint)
     }
-    return textPaint!!
   }
 
   companion object {
     private val INSTANCE = XRay()
     private const val FRAME_WIDTH = 500
     private const val FRAME_HEIGHT = 150
-    private const val XRAY_ALFA = 0.9f
+    private const val XRAY_ALPHA = 0.9f
     private const val TEXT_SIZE = 30
     private const val TEXT_COLOR = Color.RED
 
-    /** Toggles state of XRay. */
+    /** Setup XRay using a [XRayConfig] */
     @JvmStatic
-    fun toggle() {
-      INSTANCE.isEnabled = !INSTANCE.isEnabled
+    public fun setup(config: XRayConfig) {
```

Comment:
Do we need thread-safety? If so, we must instead provide a way to atomically "read and write" -- an `.update { }` function, as below:

```
INSTANCE = AtomicReference<XRayConfig>(XRayConfig())

inline fun update(function: (current: T) -> T) {
  INSTANCE.getAndUpdate(function)
}
```

Reply:
I think we don´t need thread-safety. At least I never had thread issues with XRay.
Is it ok for you if we let it as is?

Reply:
Yes it's ok, was just wondering.